tests/vmi_cloudinit_test.go

```diff
@@ -438,14 +437,18 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 			})
 			It("[test_id:4622]should have cloud-init meta_data with tagged devices", func() {
 				testInstancetype := "testInstancetype"
-				vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdataNetworkData(
-					cd.ContainerDiskFor(cd.ContainerDiskCirros), "", testNetworkData, false)
-				vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{{Name: "default", Tag: "specialNet", InterfaceBindingMethod: v1.InterfaceBindingMethod{Masquerade: &v1.InterfaceMasquerade{}}}}
-				vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
-				if vmi.Annotations == nil {
-					vmi.Annotations = make(map[string]string)
-				}
-				vmi.Annotations[v1.InstancetypeAnnotation] = testInstancetype
+				vmi := libvmifact.NewCirrosNoUserData(
+					libvmi.WithCloudInitConfigDriveNetworkData(testNetworkData),
+					libvmi.WithInterface(v1.Interface{
+						Name: "default",
+						Tag:  "specialNet",
+						InterfaceBindingMethod: v1.InterfaceBindingMethod{
+							Masquerade: &v1.InterfaceMasquerade{},
+						},
+					}),
```

Comment:
nit: I would keep it inline or emphesize the test subject field `tag`:
```suggestion
libvmi.WithInterface(v1.Interface{Name: "default", InterfaceBindingMethod: v1.InterfaceBindingMethod{Masquerade: &v1.InterfaceMasquerade{},
        Tag:  "specialNet",
}}),
```